### DISCUSSION TITLE
How to include directories in all CMake targets

### DISCUSSION THREAD

**Discussion by JackNWhite** [nasa/fprime#1449] [Q&A] [answered]
Hi,

I have a hand-written serializable which is built as part of our project's types directory. This class includes a number of headers from a second project. I have included these headers in the types directory CMakeLists using `include_directories()`, but any other component or port which depends on the new class just includes its headers, so the second-project directories are not included.

How do I ensure these directories are always included, so I don't have to add a long list of inclusions to multiple CMakeLists across the project?

**Comment by LeStarch** [comment, marked as answer]
Move the `include_directories()` call to the top of the `CMakeList.txt` where you call `project(...`.  This is the deployment CMake and will effectively make these include directories global.

You could also switch to using `target_include_directories(<module> PUBLIC ....)` to make them public headers of the target itself, but this will likely take more work to iron out the first time you do it.
